pkg/epp/flowcontrol/registry/registry.go

```diff
@@ -156,135 +153,137 @@ func NewFlowRegistry(config Config, logger logr.Logger, opts ...RegistryOption)
-	events := make(chan event, config.EventChannelBufferSize)
-
 	fr := &FlowRegistry{
 		config:               validatedConfig,
 		logger:               logger.WithName("flow-registry"),
-		flowStates:           make(map[types.FlowKey]*flowState),
-		events:               events,
 		activeShards:         []*registryShard{},
 		drainingShards:       make(map[string]*registryShard),
 		perPriorityBandStats: make(map[uint]*bandStats, len(validatedConfig.PriorityBands)),
-		// Initialize `generation` to 1. A generation value of 0 in `flowState` is reserved as a sentinel value to indicate a
-		// brand new flow that has not yet been processed by the GC loop.
-		gcGeneration: 1,
 	}
 
 	for _, opt := range opts {
 		opt(fr)
 	}
-
 	if fr.clock == nil {
 		fr.clock = &clock.RealClock{}
 	}
 
-	fr.gcTicker = fr.clock.NewTicker(fr.config.FlowGCTimeout)
-
 	for i := range config.PriorityBands {
 		band := &config.PriorityBands[i]
 		fr.perPriorityBandStats[band.Priority] = &bandStats{}
 	}
 
-	// `UpdateShardCount` handles the initial creation and populates `activeShards`.
-	if err := fr.UpdateShardCount(validatedConfig.InitialShardCount); err != nil {
-		fr.gcTicker.Stop()
+	if err := fr.updateShardCount(validatedConfig.InitialShardCount); err != nil {
 		return nil, fmt.Errorf("failed to initialize shards: %w", err)
 	}
-
 	fr.logger.V(logging.DEFAULT).Info("FlowRegistry initialized successfully")
 	return fr, nil
 }
 
-// Run starts the registry's background event processing loop. It blocks until the provided context is cancelled.
-// This loop implements the serialized control plane (Actor model), handling both asynchronous signals from data plane
-// components and periodic ticks for garbage collection.
+// Run starts the registry's background garbage collection loop.
+// It blocks until the provided context is cancelled.
 func (fr *FlowRegistry) Run(ctx context.Context) {
-	fr.logger.Info("Starting FlowRegistry event loop")
-	defer fr.logger.Info("FlowRegistry event loop stopped")
-	defer fr.gcTicker.Stop()
+	fr.logger.Info("Starting FlowRegistry background garbage collection loop")
+	defer fr.logger.Info("FlowRegistry background garbage collection loop stopped")
+	gcTicker := fr.clock.NewTicker(fr.config.FlowGCTimeout)
+	defer gcTicker.Stop()
 
 	for {
 		select {
 		case <-ctx.Done():
 			return
-		case <-fr.gcTicker.C():
-			fr.mu.Lock()
-			fr.onGCTick()
-			fr.mu.Unlock()
-		case evt := <-fr.events:
-			fr.mu.Lock()
-			switch e := evt.(type) {
-			case *queueStateChangedEvent:
-				fr.onQueueStateChanged(e)
-			case *shardStateChangedEvent:
-				fr.onShardStateChanged(e)
-			case *syncEvent:
-				close(e.doneCh) // Signal synchronization point reached.
-			}
-			fr.mu.Unlock()
+		case <-gcTicker.C():
+			fr.executeGCCycle()
 		}
 	}
 }
 
-// RegisterOrUpdateFlow handles the registration of a new flow instance or the update of an existing instance's
```

Comment:
Removed in favor of the JIT registration in the `WithConnection` path. We don't allow flow specific config overrides yet, so a`Reconcile/Update` method is not necessary.